tests/gar.test.mjs

```diff
@@ -775,329 +510,533 @@ describe('GatewayRegistry', async () => {
           observedEpochCount: 0,
         },
       });
+
+      if (delegateAddresses && expectedDelegates) {
+        var nextMemory = joinNetworkMemory;
+        for (const delegateAddress of delegateAddresses) {
+          const { memory: delegatedMemory } = await delegateStake({
+            memory: joinNetworkMemory,
+            timestamp: STUB_TIMESTAMP,
+            delegatorAddress: delegateAddress,
+            quantity: 500_000_000,
+            gatewayAddress: otherGatewayAddress,
+          }).catch(() => {});
+          nextMemory = delegatedMemory;
+        }
+        const updatedGateway = await getGateway({
+          address: otherGatewayAddress,
+          memory: nextMemory,
+        });
+        assert.deepEqual(
+          Object.keys(updatedGateway.delegates),
+          expectedDelegates,
+        );
+      }
+    }
+
+    it('should allow joining of the network with an allow list', async () => {
+      await allowlistJoinTest({
+        tags: [
+          { name: 'Allow-Delegated-Staking', value: 'allowlist' },
+          { name: 'Allowed-Delegates', value: STUB_ADDRESS }, // todo: exercise comma separated list
+        ],
+        expectedAllowDelegatedStaking: true,
+        expectedAllowedDelegatesLookup: {
+          [STUB_ADDRESS]: true,
+        },
+        delegateAddresses: [STUB_ADDRESS],
+        expectedDelegates: [STUB_ADDRESS],
+      });
+    });
+
+    it('should join the network ignoring the allow list if Allow-Delegated-Staking is set to "true"', async () => {
+      await allowlistJoinTest({
+        tags: [
+          { name: 'Allow-Delegated-Staking', value: 'true' },
+          { name: 'Allowed-Delegates', value: STUB_ADDRESS },
+        ],
+        expectedAllowDelegatedStaking: true,
+      });
+    });
+
+    it('should join the network ignoring the allow list if Allow-Delegated-Staking is set to "false"', async () => {
+      await allowlistJoinTest({
+        tags: [
+          { name: 'Allow-Delegated-Staking', value: 'false' },
+          { name: 'Allowed-Delegates', value: STUB_ADDRESS },
+        ],
+        expectedAllowDelegatedStaking: false,
+      });
+    });
+
+    it('should join the network and start an empty allow list if Allow-Delegated-Staking is set to "allowlist"', async () => {
+      await allowlistJoinTest({
+        tags: [{ name: 'Allow-Delegated-Staking', value: 'allowlist' }],
+        expectedAllowDelegatedStaking: true,
+        expectedAllowedDelegatesLookup: [], // TODO: Why is an empty Lua table JSON serialized to an array?
```

Comment:
it's part of json.lua - without any objects, there's no way to clarify if you want a json object, or table back.

Reply:
Yep, figured this was the case!